sbin/send_email.py

```diff
@@ -124,27 +148,55 @@ def send_email():
     db = setup_db()
     queued_emails = get_email_queue(db)
     mail_client = construct_mail_client()
-
     if len(queued_emails) == 0:
         return
 
+    success_count = 0
+
     for email_data in queued_emails:
+        if email_data["send_to"] == "":
+            store_error(email_data["id"], db, "ERROR: empty email address")
+            e = "[{}] ERROR: empty email address for recipient {}".format(
+                str(datetime.datetime.now()), email_data["user_id"])
+            LOG_FILE.write(e+"\n")
+            print(e)
+            continue
+
         email = construct_mail_string(
             email_data["send_to"], email_data["subject"], email_data["body"])
-        mail_client.sendmail(EMAIL_SENDER, email_data["send_to"], email.encode('utf8'))
-        mark_sent(email_data["id"], db)
 
-    LOG_FILE.write("[{}] Sucessfully Emailed {} Users\n".format(
-        str(datetime.datetime.now()), len(queued_emails)))
+        try:
+            mail_client.sendmail(EMAIL_SENDER,
+                                 email_data["send_to"], email.encode('utf8'))
+            mark_sent(email_data["id"], db)
+            success_count += 1
+
+        except Exception as email_send_error:
+            store_error(email_data["id"], db, "ERROR: sending email "
+                        + str(email_send_error))
+            e = "[{}] ERROR: sending email to recipient {}, email {}: {}".format(
+                str(datetime.datetime.now()),
+                email_data["user_id"],
+                email_data["send_to"],
+                str(email_send_error))
+            LOG_FILE.write(e+"\n")
+            print(e)
+
+    e = "[{}] Sucessfully Emailed {} Users".format(
+        str(datetime.datetime.now()), success_count)
+    LOG_FILE.write(e+"\n")
+    print(e)
```

Comment:
This shouldn't print anything on a successful run as any output will be sent to the owner of the crontab/value of MAILTO, regardless of its on sysout or syserr.